packages/core/src/amazonq/webview/messages/messageDispatcher.ts

```diff
@@ -10,11 +10,53 @@ import { TabType } from '../ui/storages/tabsStorage'
 import { getLogger } from '../../../shared/logger'
 import { amazonqMark } from '../../../shared/performance/marks'
 import { telemetry } from '../../../shared/telemetry'
+import { TelemetrySpan } from '../../../shared/telemetry/spans'
+import { ChatSessionStorageFacade } from '../../util/storageFacade'
+import { AmazonQUITelemetryHelper } from '../../util/telemetryHelper'
+import { globals } from '../../../shared'
+import { MetadataEntry, MetricDatum } from '../../../shared/telemetry/clienttelemetry'
 
 export function dispatchWebViewMessagesToApps(
     webview: Webview,
-    webViewToAppsMessagePublishers: Map<TabType, MessagePublisher<any>>
+    webViewToAppsMessagePublishers: Map<TabType, MessagePublisher<any>>,
+    chatSessionFacade: ChatSessionStorageFacade
 ) {
+    /**
+     * Listen to all incoming telemetry events. If one of them has cwsprChatConversationId or amazonQConversationId
+     * then it is considered an Amazon Q event that we want to track.
+     *
+     * We build up all the Amazon Q events we want to track by conversation ID
+     *
+     * This allows us to eventually connect metadata coming from a conversation ID with metadata coming from a tabID. Tabs know nothing
+     * of conversation IDs and conversation IDs have no knowledge of tabIDs except for in chat session storages.
+     *
+     * In the flow Webview -> Message Dispatcher -> Amazon Q Feature -> Amazon Q Feature Implementation -> Message Dispatcher -> Webview
+     * only "Amazon Q Feature Implementation" has knowledge about conversation IDs
+     * everything else has knowledge of tabIDs
+     */
+    telemetry.chat_roundTrip.listen(
```

Comment:
nit: This is a stylistic preference but would it make sense to gather all the RTT related code in this file in to a class such as:

```
class QChatMessageDuration {
  static initializeListener()
  static startListening()
  static stopListening()
}
```

and then on line 37, 80 and 86 you can swap that whole implementation with the function call.

----

My reasoning is that if someone needs to add something else to the general functions there is less to parse